### ISSUE TITLE
[BUG]: Blank Lines in Executor Output are Omitted when Following a `\0` Char

### ISSUE THREAD

**Issue by mgerhold** [compiler-explorer/compiler-explorer#8152]
### Describe the bug

Given this program:
```cpp
#include <iostream>

int main() {
    std::cout << "a\n";
    std::cout << '\0' << '\n';
    std::cout << "b\n";
}
```
It yields this output:
```
a
b
```
But there should be a blank line between `a` and `b`. However, if `\n` is changed to `\r`, there *is* a blank line.

https://godbolt.org/z/hqf97a5qM

### Steps to reproduce

1. Visit https://godbolt.org/z/hqf97a5qM
2. Look at execution output and compare with expected output

### Expected behavior

Expected output is:
```
a

b
```

### Reproduction link

https://godbolt.org/z/hqf97a5qM

### Screenshots

<img width="1903" height="359" alt="Image" src="https://github.com/user-attachments/assets/479908a5-0e38-4afc-95df-a8039e90be26" />

### Operating System

Windows 11

### Browser version

Firefox 143.0.1 (64-Bit)

**Comment by mattgodbolt** [comment]
We decode the output as UTF-8. I don't think that's valid UTF8, which might be the issue here.

**Comment by mgerhold** [comment]
If I understand it correctly, ASCII (range 0 to 127) is a valid subset of UTF-8.

Also, this little Python program here does not complain about the sequence of 0x00, 0x0A (which is null terminator + line feed):
```python
def print_decoded_bytes_quoted(data: bytes) -> None:
    text = data.decode("utf-8")
    print(f'"{text}"')

print_decoded_bytes_quoted(b"\x00\x0A")
```
Output (both locally and [in Compiler Explorer](https://godbolt.org/z/6WfjeM3E7)):
```
"
"
```

**Comment by mattgodbolt** [comment]
Thanks! We'll take a look

**Comment by Kasalop** [comment]
The error seems to be right here: https://github.com/compiler-explorer/compiler-explorer/blob/dc730bd2ac633630a71d02375aa746b4270e5b69/static/panes/executor.ts#L540

The check is an equal check with an empty string, not an "emptyish" string. So an empty div is generated (that collapses to a div with zero height) instead of the needed `<div><br/></div>`.


edit: Another solution would be to add a token replacement in ansi-to-html.ts: https://github.com/compiler-explorer/compiler-explorer/blob/main/static/ansi-to-html.ts#L396

Even if not 100% true, you could use the unicode `\u2400` (␀) to represent that it's the `\0` char. I personal would prefer that, to not get confused with an actual empty string. One way or the other there is a duplication for the output:

- \0 gets replaced with ␀ : `\0` and `\c2400` both produce ␀ as output
- \0 gets replaced with <empty string>: `\0` and `` (an actual empty string) both produce '' as output

**Comment by mattgodbolt** [comment]
Thanks @Kasalop ! Not at all where I would have expected it to be!
